numpy/random/mtrand/mtrand.pyx

```diff
@@ -1254,17 +1580,62 @@ cdef class RandomState:
         key = np.dtype(dtype).name
         if not key in _randint_type:
             raise TypeError('Unsupported dtype "%s" for randint' % key)
-        lowbnd, highbnd, randfunc = _randint_type[key]
+        lowbnd, highbnd, func = _randint_type[key]
+
+        if np.array(low).shape == np.array(high).shape == ():
+            # Determine if `low` and `high` are both scalars
+            # Otherwise, at least one of them is array_like
+            #
+            # These conversions also ensure that we perform
+            # the subsequent comparisons with Python integers
+            # and not with Numpy integer types, which are
+            # susceptible to an unusual corner case described
+            # below under the EXCEPT-block
+            low = int(low)
+            high = int(high)
+
+            if low < lowbnd:
+                raise ValueError("low is out of bounds for %s" % (key,))
+            if high > highbnd:
```

Comment:
This should be `>=`, not `>`, `highbnd` is the largest possible value for high plus one.